homeassistant/components/sensor/openhardwaremonitor.py

```diff
@@ -0,0 +1,192 @@
+"""Support for Open Hardware Monitor Sensor Platform."""
+
+from datetime import timedelta
+import logging
+import requests
+import voluptuous as vol
+
+from homeassistant.util.dt import utcnow
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_MIN_VALUE = 'minimal_value'
+STATE_MAX_VALUE = 'maximum_value'
+STATE_VALUE = 'value'
+STATE_OBJECT = 'object'
+CONF_INTERVAL = 'interval'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+SCAN_INTERVAL = timedelta(seconds=30)
+RETRY_INTERVAL = timedelta(seconds=30)
+
+OHM_VALUE = 'Value'
+OHM_MIN = 'Min'
+OHM_MAX = 'Max'
+OHM_CHILDREN = 'Children'
+OHM_NAME = 'Text'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=8085): cv.port
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Open Hardware Monitor platform."""
+    data = OpenHardwareMonitorData(config, hass)
+    add_devices(data.devices, True)
```

Comment:
If you have to retry, this call will just add nothing and nothing will ever be added.